src/components/block.mjs

```diff
@@ -0,0 +1,40 @@
+// @flow
+
+import crypto from 'crypto';
+
+import type Transaction from './transaction';
+
+class Block {
+  id: string;
+  transactions: Transaction[];
+  previousHash: string;
+  timestamp: number;
+
+  constructor(previousHash: string = '') {
+    this.id = crypto.randomBytes(32).toString('hex');
+    this.transactions = [];
+    this.previousHash = previousHash;
+    this.timestamp = Date.now();
+  }
+
+  async computeHash(): Promise<string> {
```

Comment:
We should consider using sha256 instead of md5, slower (~60%) but more secure as it avoids collisions in hash generation. 
The rest is great!
Although this will need adjustment when we implement mining features. The mining is a nonstop process. If the compute hashing method is async we may halt the system very frequently, thus it would mess with incoming transactions and communication with other node.

Reply:
We should allow concurrency between nodes